### PR TITLE
[FINAL] Unity compatibility

### PR DESCRIPTION
**Pull request by vegaro**
Dates in Unity were using millis instead of Strings. I basically copied over all the dates from the Unity files and changed the name to end with "Millis".

The cons of these changes will also be present in the Javascript objects. RN and cordova converts anything we send to the Javascript layer automatically to Javascript objects. I don't think it's a big deal since we have well defined Typescript types.

### CODE REVIEW

File: android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt [hunk 1]
```diff
@@ -17,19 +17,28 @@ fun EntitlementInfo.map(): Map<String, Any?> =
         "isActive" to this.isActive,
         "willRenew" to this.willRenew,
         "periodType" to this.periodType.name,
+        "latestPurchaseDateMillis" to this.latestPurchaseDate.time / 1000.0,
```
Comment (aboedo) [inline android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt:20]: maybe a Date extension would make cleaner by killing duplication as in `toMillis()`

Comment (vegaro) [reply android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt:20]: yeah let's do it

File: android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt [hunk 2]
```diff
@@ -222,3 +231,35 @@ private fun String?.mapPeriod(): Map<String, Any?> {
         }
     }
 }
+
+fun Map<String, *>.convertToJson(): JSONObject {
```
Comment (aboedo) [inline android/common/src/main/java/com/revenuecat/purchases/common/mappers.kt:235]: 😍

File: ios/NSDate+HybridAdditions.m [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//
+// Created by Andrés Boedo on 3/4/20.
```
Comment (aboedo) [inline ios/NSDate+HybridAdditions.m:2]: LOL I don't think I created this one

**Pull request merged**